### PR TITLE
feature: add TMS_Provider

### PR DESCRIPTION
**Pull request by peppsac**
Add TMS_Provider and a simple example to display OSM map à la OpenLayers

### CODE REVIEW

**Comment by peppsac** [comment]
Added an example that uses the `TMS_Provider`.

File: examples/orthographic.html [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+        </style>
+        <meta charset="UTF-8">
+
+        <meta name="viewport" content="width=device-width, initial-scale=1.0">
+    </head>
+    <body>
+        <div id="viewerDiv"></div>
+        <script src="/dist/itowns.js"></script>
+        <script type="text/javascript">
+            let renderer;
```
Comment (autra) [inline examples/orthographic.html:24]: Why do you need this here?

File: examples/orthographic.js [hunk 2]
```diff
@@ -0,0 +1,92 @@
+/* global itowns, document, renderer */
```
Comment (autra) [inline examples/orthographic.js]: If you do `let renderer` here, you don't need it in global. Also I'm suprised to see `document` here. eslint knows about `document` being global :-)

File: examples/orthographic.js [hunk 3]
```diff
@@ -0,0 +1,92 @@
...
+
+const r = viewerDiv.clientWidth / viewerDiv.clientHeight;
+const camera = new itowns.THREE.OrthographicCamera(
+    extent.west(), extent.east(),
+    extent.east() / r, extent.west() / r,
+    0, 1000);
+
+// Instanciate PlanarView
+var view = new itowns.PlanarView(viewerDiv, extent, { renderer, maxSubdivisionLevel: 10, camera });
+view.tileLayer.disableSkirt = true;
```
Comment (autra) [inline examples/orthographic.js]: I think this `disableSkirt` is not obvious to the newcomers. Can you add a comment to explain why it's needed?

File: src/Core/Geographic/Extent.js [hunk 4]
```diff
@@ -13,14 +13,15 @@ const CARDINAL = {
     NORTH: 3,
 };
 
-function _crsIsWMTS(crs) {
-    return crs.indexOf('WMTS:') == 0;
+function _crsIsWMTSorTMS(crs) {
```
Comment (autra) [inline src/Core/Geographic/Extent.js]: `_crsIsTiled`?

File: src/Core/Scheduler/Providers/OGCWebServiceHelper.js [hunk 5]
```diff
@@ -98,6 +98,22 @@ export default {
                 projection.getCoordWMTS_WGS84(tileCoord, tile.extent, tileMatrixSet);
         }
     },
+    computeTMSCoordinates(tile, extent) {
```
Comment (autra) [inline src/Core/Scheduler/Providers/OGCWebServiceHelper.js:101]: A bit of doc?

File: src/Core/Scheduler/Providers/OGCWebServiceHelper.js [hunk 6]
```diff
@@ -98,6 +98,22 @@ export default {
...
+        if (tile.extent.crs() != extent.crs()) {
+            throw new Error('Unsupported configuration. TMS is only supported when geometry has the same crs than TMS layer');
+        }
+        const c = tile.extent.center();
+        const layerDimension = extent.dimensions();
+
+        // normalize x/y
+        const x = (1 + c.x() / (layerDimension.x * 0.5)) * 0.5;
+        const y = (1 - c.y() / (layerDimension.y * 0.5)) * 0.5;
```
Comment (autra) [inline src/Core/Scheduler/Providers/OGCWebServiceHelper.js]: does this suppose that the origin is at the center of the extent? Is it a correct assumption?

File: src/Core/Scheduler/Providers/TMS_Provider.js [hunk 7]
```diff
@@ -0,0 +1,69 @@
...
+        throw new Error(`Missing extent property for layer '${layer.id}'`);
+    }
+    if (!layer.projection) {
+        throw new Error(`Missing projection property for layer '${layer.id}'`);
+    }
+    layer.extent = new Extent(layer.projection, ...layer.extent);
+    if (!layer.options.zoom) {
+        layer.options.zoom = {
+            min: 0,
+            max: 18,
```
Comment (autra) [inline src/Core/Scheduler/Providers/TMS_Provider.js:19]: `getCapabilities`... I forgot about this one. I need to resume the work on that.

File: src/Core/Scheduler/Providers/TMS_Provider.js [hunk 8]
```diff
@@ -0,0 +1,69 @@
...
+    /* eslint-enable no-template-curly-in-string */
+};
+
+TMS_Provider.prototype.executeCommand = function executeCommand(command) {
+    const layer = command.layer;
+    const tile = command.requester;
+    const coordTMS = tile.getCoordsForLayer(layer)[0];
+    const coordTMSParent = (command.targetLevel < coordTMS.zoom) ?
+        OGCWebServiceHelper.WMTS_WGS84Parent(coordTMS, command.targetLevel) :
+        undefined;
```
Comment (autra) [inline src/Core/Scheduler/Providers/TMS_Provider.js:38]: So this is : "if the requested level is less than my current level, then let's find out which one of my parents is on this level". That's it? 

How is it possible that the requester ask for a level lower than its own? For some network strategy like dichotomy?

**Review by autra — CHANGES_REQUESTED** [review]

File: examples/orthographic.html (hunk 1 above)
Comment (peppsac) [reply examples/orthographic.html:24]: It's not needed yet... it's only needed to write a test using this example.

File: src/Core/Scheduler/Providers/TMS_Provider.js (hunk 8 above)
Comment (peppsac) [reply src/Core/Scheduler/Providers/TMS_Provider.js:38]: Yes. And Yes. :)

File: src/Core/Scheduler/Providers/OGCWebServiceHelper.js (hunk 6 above)
Comment (peppsac) [reply src/Core/Scheduler/Providers/OGCWebServiceHelper.js]: There's no particular assumption, except that `c` is inside the `layer`

**Comment by peppsac** [comment]
Thanks @autra 
I've updated the branch.

File: src/Core/Scheduler/Providers/OGCWebServiceHelper.js (hunk 6 above)
Comment (peppsac) [reply src/Core/Scheduler/Providers/OGCWebServiceHelper.js]: Ok understood what you meant, and I rewrote the code to be clearer (I hope).

**Review by autra — APPROVED** [review]

**Comment by autra** [comment]
LGTM, thanks! Please rebase :-)

**Pull request merged**